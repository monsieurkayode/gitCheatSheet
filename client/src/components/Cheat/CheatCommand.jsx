import React from 'react';
import PropTypes from 'prop-types';

const CheatCommand = ({ command }) => {
  const [keyword, ...body] = command.split(' ');

  return (
    <pre className="command">
      <code>&#36;&nbsp;
        <span className="git">{keyword}</span>&nbsp;{body.join(' ')}
      </code>
    </pre>
  );
};

CheatCommand.propTypes = {
  command: PropTypes.string.isRequired
};

export default CheatCommand;
