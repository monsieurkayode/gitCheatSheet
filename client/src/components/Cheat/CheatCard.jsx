import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import CheatCommand from './CheatCommand';

const Container = styled.div`
  margin: 0 2%;
  min-width: 300px;
  padding-bottom: 20px;
  break-inside: avoid;
    

  header {
    display: flex;
    flex-flow: row nowrap;
    justify-content: space-between;
    align-items: center;

    .heading {
      color: #6756b3;
      font-size: 18px;
      padding: 14px 0;
      font-family: Roboto, sans-serif;
      font-weight: 400;

      @media only screen and (max-width: 768px) {
        font-size: 16px;
      }
    }

    &::after {
      content: "";
      flex-grow: 1;
      border-bottom: 1px solid transparent;
      border-image-source: linear-gradient(
        90deg, rgba(0,0,0,0.2) 20%, rgba(0,0,0,0.05) 80%);
      border-image-slice: 1;
      margin-left: 24px;
    }
  }

  .body-area {
    box-shadow: inset 1px 4px 20px 0px rgba(0,0,0,0.1),
      1px 4px 20px -4px rgba(0,0,0,0.1),
      -1px 3px 20px -4px rgba(0,0,0,0.1);
  
    .commands {
      background: #FFF;
      padding: 16px 20px;

      .command {
        ::-webkit-scrollbar {
          width: 0;
        }

        overflow: auto;
        font-family: Cousine, sans-serif;
        line-height: 20px;
        font-size: 14px;
        cursor: pointer;

        .git {
          color: teal;
        }

        @media only screen and (max-width: 768px) {
          font-size: 12px;
        }
      }
    }

    .description {
      background: #f5f6f7;
      padding: 12px 20px;
      font-size: 14px;
      font-family: Prompt;
      font-weight: 100;
      color: #668090;

      @media only screen and (max-width: 768px) {
        font-size: 12px;
      }
    }
  }
`;

const CheatCard = ({
  commands,
  description,
  header
}) => (
  <Container>
    <header>
      <h6 className="heading">{header}</h6>
    </header>
    <main className="body-area">
      <div className="commands">
        {commands
          .map(command => <CheatCommand key={command} command={command} />)}
      </div>
      <div className="description">{description}</div>
    </main>
  </Container>
);

CheatCard.propTypes = {
  commands: PropTypes.arrayOf(PropTypes.string).isRequired,
  description: PropTypes.string.isRequired,
  header: PropTypes.string.isRequired
};

export default CheatCard;
