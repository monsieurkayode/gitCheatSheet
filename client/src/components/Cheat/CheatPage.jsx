/* eslint-disable no-underscore-dangle */
import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import CheatCard from './CheatCard';
import NoContent from '../Common/NoContent';
import Loader from '../Common/Loader';

const Container = styled.main`
  max-width: 1232px;
  margin-top: 60px;
  padding: 0 1%;
  flex: 1;
  width: 100%;
`;

const HeaderWrapper = styled.div`
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
`;

const Heading = styled.h2`
  font-family: 'Josefin Sans', sans-serif;
  font-weight: 300;
  font-size: 40px;
  text-transform: capitalize;
  text-align: center;
  color: ${({ sub }) => (sub ? '#668090' : '')};
  padding: 4px 0;

  @media only screen and (max-width: 768px) {
    font-size: 34px;
  }
`;

const Section = styled.section`
  column-count: auto;
  column-width: 360px;
  margin-top: 20px;
`;

const CheatPage = ({ match, cheats, loading }) => (
  <Fragment>
    {cheats.length === 0 && (
      <Loader
        style={{ display: loading ? '' : 'none' }}
        size={40}
        strokeColor="#6756b3"
        loading={loading}
      />
    )}
    {cheats.length === 0 && !loading && (
    <NoContent
      content="Oops, sorry can't find what you're looking for"
    />)}
    {cheats.length > 0 && (
    <Container>
      <HeaderWrapper>
        <Heading>{match.params.category.replace(/-/g, ' ')}&nbsp;</Heading>
        <Heading sub>cheatsheet</Heading>
      </HeaderWrapper>
      <Section>
        {cheats.map(cheat => <CheatCard key={cheat._id} {...cheat} />)}
      </Section>
    </Container>
    )}
  </Fragment>
);

CheatPage.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      category: PropTypes.string.isRequired
    }).isRequired
  }).isRequired,
  cheats: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  loading: PropTypes.bool.isRequired
};

const mapStateToProps = ({ cheatSheets }, { match }) => {
  let { category } = match.params;
  category = category.replace(/-/g, ' ');
  const cheats = cheatSheets.cheats
    .filter(cheat => cheat.category.name === category);
  return { cheats, loading: cheatSheets.makingApiRequest };
};

export default connect(mapStateToProps)(CheatPage);
