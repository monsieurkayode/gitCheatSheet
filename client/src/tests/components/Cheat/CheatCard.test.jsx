import React from 'react';
import { shallow } from 'enzyme';
import { CheatCard } from '../../../components/Cheat/CheatCard';

const setup = () => {
  const props = {
    commands: ['git merge'],
    description: 'desc',
    header: 'header',
    _id: '1',
    setId: jest.fn(),
    copiedId: '1'
  };

  return shallow(<CheatCard {...props} />);
};

describe('CheatCard Component', () => {
  const wrapper = setup();

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
