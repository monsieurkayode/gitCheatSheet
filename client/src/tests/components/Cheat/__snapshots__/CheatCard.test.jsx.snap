// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheatCard Component should render correctly 1`] = `
<styled.div>
  <header>
    <h6
      className="heading"
    >
      header
    </h6>
  </header>
  <main
    className="body-area"
  >
    <span
      className="tip"
    >
      copied!
    </span>
    <div
      className="commands"
    >
      <CheatCommand
        command="git merge"
        id="1"
        key="git merge"
        onCopy={[MockFunction]}
      />
    </div>
    <div
      className="description"
    >
      desc
    </div>
  </main>
</styled.div>
`;
